tests/test_iqm_job.py

```diff
@@ -70,24 +70,20 @@ def test_status_for_ready_result(job):
     assert result.get_memory() == ['11', '10', '10']
 
 
-def test_status_done(job, iqm_result_single_register):
-    client_result = RunResult(status=RunStatus.READY, measurements=iqm_result_single_register)
-    when(job._client).get_run(uuid.UUID(job.job_id())).thenReturn(client_result)
+def test_status_done(job):
+    client_result = RunResult(status=RunStatus.READY, measurements=None)
+    when(job._client).get_run_status(uuid.UUID(job.job_id())).thenReturn(client_result)
     assert job.status() == JobStatus.DONE
-    assert job._result is not None
-
-    # Assert that repeated call does not query the client (i.e. works without calling the mocked get_run)
-    assert job.status() == JobStatus.DONE
-    mockito.verify(job._client, times=1).get_run(uuid.UUID(job.job_id()))
```

Comment:
@q-mat-beu These two lines relied on the fact that `IQMJob.status()` called `get_run` and saved the results (in memory) if job completed. Thus, if you called `IQMJob.status()` and got `DONE`, then calling it again would return `DONE` immediately without a client call, thanks to [this line](https://github.com/iqm-finland/qiskit-on-iqm/blob/main/src/qiskit_iqm/iqm_job.py#L100).

Since updated `IQMJob.status()` calls `get_run_status`, it does not save the results. Thus `IQMJob.status()` *always* calls the client. So, the only way to adapt those two lines is to change `times=1` to `times=2`, which I'm not sure is useful.

Reply:
Makes sense. I thought the status might be cached, too.

Reply:
Hmm... I haven't thought of caching the status. It'd make sense if we want to optimize this sort of communication, but, after a brief discussion with @hay-k, it seems it's not worth it, and requesting the status isn't really expensive.